src/components/common/ProductBox/ProductBox.js

```diff
@@ -61,8 +62,13 @@ const ProductBox = ({
       <div className={styles.line}></div>
       <div className={styles.actions}>
         <div className={styles.outlines}>
-          <Button variant='outline'>
-            <FontAwesomeIcon icon={faHeart} className={heart ? styles.heart : ''}>
+          <Button variant='outline' onClick={() => addToFavourite(id)}>
```

Comment:
Potrzebujesz tutaj stworzyć metodę do obsługi kliknięcia, analogicznie jak `compareHandler` aby wywołać `event.preventDefault` , bo w tej chwili jak klikniemy na przycisk to strona skacze.